Makefile

```
@@ -0,0 +1,120 @@
+#
+# This file is a template Makefile. Some targets are presented here as examples.
+# Feel free to customize it to your needs!
+#
+DOCKER_COMPOSE = docker-compose
+CMD_ON_PROJECT = $(DOCKER_COMPOSE) run -u www-data --rm php
+PHP_RUN = $(CMD_ON_PROJECT) php
+YARN_RUN = $(DOCKER_COMPOSE) run -u node --rm -e YARN_REGISTRY -e PUPPETEER_SKIP_CHROMIUM_DOWNLOAD node yarn
+CONSOLE = $(PHP_RUN) bin/console
+AKENEO_FIXTURES = vendor/akeneo/pim-community-dev/src/Akeneo/Platform/Bundle/InstallerBundle/Resources/fixtures
+
+ifdef NO_DOCKER
+  CMD_ON_PROJECT =
+  YARN_RUN = yarnpkg
+  PHP_RUN = php
+endif
+
+.DEFAULT_GOAL := dev
+
+yarn.lock: package.json
+	PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=1 $(YARN_RUN) install
+
+node_modules: yarn.lock
+	PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=1 $(YARN_RUN) install
+
+.PHONY: assets
+assets:
+	$(CMD_ON_PROJECT) rm -rf public/bundles public/js
+	$(CONSOLE) pim:installer:assets --symlink --clean
+
+.PHONY: css
+css:
+	$(CMD_ON_PROJECT) rm -rf public/css
+	$(YARN_RUN) run less
+
+.PHONY: javascript-prod
+javascript-prod:
+	$(CMD_ON_PROJECT) rm -rf public/dist
+	$(YARN_RUN) run webpack
+
+.PHONY: javascript-dev
+javascript-dev:
+	$(CMD_ON_PROJECT) rm -rf public/dist
+	$(YARN_RUN) run webpack-dev
+
+.PHONY: front
+front: assets css javascript-dev
+
+.PHONY: database
+database:
+	echo "This would have deleted the database entirely and recreated it empty"
+	echo "Uncomment out the below line to actually run the task"
+	exit 1
+	# $(CONSOLE) pim:installer:db ${O}
+
+.PHONY: cache
+cache:
+	$(CMD_ON_PROJECT) rm -rf var/cache && $(CONSOLE) --verbose cache:warmup
+
+vendor: composer.lock
+	$(PHP_RUN) -d memory_limit=4G /usr/local/bin/composer install
+
+autoload:
```

Might want to mark this as phony, on the outside chance some file called `autoload` exists here (and to be clear that this target is supposed to run every time):
```suggestion
.PHONY: autoload
autoload:
```